tests/qa_automation/kernel_multipath.pm

```diff
@@ -0,0 +1,69 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Use qa_test_multipath to test multipath over iscsi
+# Maintainer: Petr Cervinka <pcervinka@suse.com>
+
+use base "qa_run";
+use strict;
+use testapi;
+use utils;
+use iscsi;
+
+sub start_testrun {
+    my $self = shift;
+
+    zypper_call("in open-iscsi qa_test_multipath");
+
+    systemctl 'start iscsid';
+    systemctl 'start multipathd';
+
+    # Set default variables for iscsi iqn and target
+    my $iqn    = get_var "ISCSI_IQN",    "iqn.2016-02.de.openqa";
+    my $target = get_var "ISCSI_TARGET", "10.0.2.1";
+
+    # Connect to iscsi server and obtain wwid for multipath configuration
+    iscsi_discovery $target;
+    iscsi_login $iqn, $target;
+    # virtio serial is too fast, system need delay to properly detect multiple paths
+    # TODO: FIX? Probably loop with check?
+    sleep 5;
+    my $wwid = script_output("multipathd -k\"show multipaths status\" | grep active | awk {'print \$1\'}");
+    iscsi_logout $iqn, $target;
+
+    # Configure test suite with proper iscsi iqn and target
+    assert_script_run("sed -i '/^TARGET_DISK=.*/c\\TARGET_DISK=\"$iqn\"' /usr/share/qa/qa_test_multipath/data/vars");
+    assert_script_run("sed -i '/^TARGET=.*/c\\TARGET=\"$target\"' /usr/share/qa/qa_test_multipath/data/vars");
+    assert_script_run("cat /usr/share/qa/qa_test_multipath/data/vars");
+
+    # Configure wwid in configuration files for each test
+    my @config_files = qw(active_active active_passive path_checker_dio path_checker_tur);
+    foreach my $config_file (@config_files) {
+        assert_script_run "sed -i '/wwid .*/c\\wwid $wwid' /usr/share/qa/qa_test_multipath/data/$config_file";
+        assert_script_run("cat /usr/share/qa/qa_test_multipath/data/$config_file");
+    }
+
+    $self->qaset_config();
+    assert_script_run("/usr/share/qa/qaset/qaset reset");
+    assert_script_run("/usr/share/qa/qaset/run/kernel-all-run.openqa");
+}
+
+sub test_run_list {
+    return qw(_reboot_off sw_multipath);
+}
+
+sub test_suite {
+    return 'regression';
```

Comment:
Interesting. What is that used for? And what does "regression" mean? Isn't *every* test in os-autoinst-distri-opensuse a "regression"-test?

Reply:
It is part of of every qa_automation/* test for tests from qa-testsuites and it can have different return values (kernel, regression,acceptance). And it seems to be a ghost from the history, according to https://github.com/os-autoinst/os-autoinst-distri-opensuse/commit/e77b442cd16b95ec3db265851152aba60e2d3ad2#diff-a1435e0088fb52c0726826996732c187L104 where its usage was removed. I used other test as a starter template...and i will try to remove it.